File: repos_estudiantes_ALVAREZ_primera_entrega/CARRERA_POZO_ALEJANDRA/Polimorfismo/Ejercicio3/Matriz.java
package Polimorfismo.Ejercicio3;

public class Matriz {
    protected float[][] matriz = new float[10][10];

    public Matriz() {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                matriz[i][j] = (i == j) ? 1.0f : 0.0f;
            }
        }
    }

    public Matriz(float[][] valores) {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                this.matriz[i][j] = valores[i][j];
            }
        }
    }

    public Matriz sumar(Matriz m) {
        Matriz resultado = new Matriz();
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                resultado.matriz[i][j] = this.matriz[i][j] + m.matriz[i][j];
            }
        }
        return resultado;
    }

    public Matriz restar(Matriz m) {
        Matriz resultado = new Matriz();
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                resultado.matriz[i][j] = this.matriz[i][j] - m.matriz[i][j];
            }
        }
        return resultado;
    }

    public boolean igual(Matriz m) {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                if (this.matriz[i][j] != m.matriz[i][j]) return false;
            }
        }
        return true;
    }

    public void mostrar() {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                System.out.print(matriz[i][j] + " ");
            }
        }
    }
}

File: repos_estudiantes_TARQUI_primera_entrega/REQUENA_OROS_ADRIAN_MARCELO/Polimorfismo/Polimorfismo/ejercicio3/Matriz.cs
using System;

namespace ejercicio3
{
	public class Matriz
	{
	    private float[,] matriz = new float[10, 10];
	
	    public Matriz()
	    {
	        for (int i = 0; i < 10; i++)
	            matriz[i, i] = 1.0f;
	    }
	
	    public Matriz(float[,] valores)
	    {
	        for (int i = 0; i < 10; i++)
	            for (int j = 0; j < 10; j++)
	                matriz[i, j] = valores[i, j];
	    }
	
	    public Matriz Sumar(Matriz otra)
	    {
	        Matriz resultado = new Matriz(new float[10,10]);
	        for (int i = 0; i < 10; i++)
	            for (int j = 0; j < 10; j++)
	                resultado.matriz[i, j] = this.matriz[i, j] + otra.matriz[i, j];
	        return resultado;
	    }
	
	    public Matriz Restar(Matriz otra)
	    {
	        Matriz resultado = new Matriz(new float[10,10]);
	        for (int i = 0; i < 10; i++)
	            for (int j = 0; j < 10; j++)
	                resultado.matriz[i, j] = this.matriz[i, j] - otra.matriz[i, j];
	        return resultado;
	    }
	
	    public bool Igual(Matriz otra)
	    {
	        for (int i = 0; i < 10; i++)
	            for (int j = 0; j < 10; j++)
	                if (this.matriz[i, j] != otra.matriz[i, j])
	                    return false;
	        return true;
	    }
	}

}
